src/System.Console/src/System/ConsolePal.Unix.cs

```diff
@@ -1106,16 +1106,14 @@ public override void Flush()
 
         internal sealed class ControlCHandlerRegistrar
         {
-            private static readonly Interop.Sys.CtrlCallback _handler = 
-                c => Console.HandleBreakEvent(c == Interop.Sys.CtrlCode.Break ? ConsoleSpecialKey.ControlBreak : ConsoleSpecialKey.ControlC);
             private bool _handlerRegistered;
 
             internal void Register()
             {
                 EnsureInitialized();
 
                 Debug.Assert(!_handlerRegistered);
-                Interop.Sys.RegisterForCtrl(_handler);
+                Interop.Sys.RegisterForCtrl(OnBreakEvent);
```

Comment:
This is going to allocate a new delegate each time it's called.  You can avoid that by putting back the static or by using a lambda here:
```C#
Interop.Sys.RegisterForCtrl(c => OnBreakEvent(c));
```

Reply:
Interesting to know. So 'OnBreakEvent` creates a delegate on the fly each time? And `c => OnBreakEvent(c)` creates a delegate once (when?) and stores it somewhere (where?)?

Reply:
> So OnBreakEvent creates a delegate on the fly each time?

Yes, unfortunately.  There's a long-standing issue to address that:
https://github.com/dotnet/roslyn/issues/5835

> And c => OnBreakEvent(c) creates a delegate once (when?) and stores it somewhere (where?)?

Yes, on first use.  Essentially it does:
```C#
private static DelegateType s_cachedDelegate;
...
DelegateType d = s_cachedDelegate ?? (s_cachedDelegate = method);
```